experimental/unitsconversion/src/converter.rs

```diff
@@ -0,0 +1,98 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use litemap::LiteMap;
+
+use crate::{
+    measureunit::MeasureUnitParser,
+    provider::{MeasureUnitItem, UnitsInfoV1},
+    ConversionError,
+};
+
+/// Represents the possible cases for the convertibility between two units.
+pub enum Convertibility {
+    /// The units are convertible.
+    /// For example, `meter` and `foot` are convertible.
+    Convertible,
+
+    /// The units are reciprocal.
+    /// For example, `meter-per-second` and `second-per-meter` are reciprocal.
+    Reciprocal,
+
+    /// The units are not convertible.
+    /// For example, `meter` and `second` are not convertible.
+    NotConvertible,
+}
+
+/// A factory for creating a converter.
+/// Also, it can check the convertibility between two units.
+pub struct ConverterFactory<'data> {
+    /// Contains the necessary data for the conversion factory.
+    payload: UnitsInfoV1<'data>,
+}
+
+impl ConverterFactory<'_> {
+    fn parser(&self) -> MeasureUnitParser<'_> {
+        MeasureUnitParser::new(&self.payload.units_conversion_trie)
+    }
+
+    /// Extract the convertibility from the given units in the form of CLDR identifiers.
+    pub fn extract_convertibility(
+        &self,
+        unit1: &str,
+        unit2: &str,
+    ) -> Result<Convertibility, ConversionError> {
+        let parser = self.parser();
+        let unit1 = parser.try_from_identifier(unit1)?;
+        let unit2 = parser.try_from_identifier(unit2)?;
+
+        struct DetermineConvertibility {
+            convertible: i16,
+            reciprocal: i16,
+        }
+
+        fn insert_units_powers(
+            unit_items: &Vec<MeasureUnitItem>,
+            sign: i8,
+            map: &mut LiteMap<u16, DetermineConvertibility>,
+        ) {
+            for item in unit_items {
+                if let Some(determine_convertibility) = map.get_mut(&item.unit_id) {
+                    determine_convertibility.convertible += (item.power * sign) as i16;
+                    determine_convertibility.reciprocal += (item.power * sign) as i16;
+                } else {
+                    map.insert(
+                        item.unit_id,
+                        DetermineConvertibility {
+                            convertible: (item.power * sign) as i16,
+                            reciprocal: (item.power * sign) as i16,
+                        },
```

Comment:
Question: It looks to me like `convertible` and `reciprocal` are calculated the exact same way. How do they differ?

Reply:
I believe line 62 should be `-=` so that equal units cancel out and we arrive at `convertible_sum == 0`?

Please write unit tests. This PR is hard to review because I have to figure out what you're trying to do, and then figure out if you're doing it correctly. It would be easier if there were unit tests to demonstrate that it's correct, then I only have to convince myself that the approach makes sense.